waddrmgr/address.go

```diff
@@ -62,11 +65,11 @@ type ManagedPubKeyAddress interface {
 	ManagedAddress
 
 	// PubKey returns the public key associated with the address.
-	PubKey() *btcec.PublicKey
+	PubKey() (*btcec.PublicKey, error)
```

Comment:
OK, we already have a db field called `compressed` on `managedAddress` which we are using for determining whether the pubkey is compressed or not. If we change the signature we might want to remove that field to avoid ambiguity.

Also, the `PubKey` signature is consistent with `PrivKey` which returns `ErrWatchingOnly` when private key is not available.